PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/AzurePlugin.java

```diff
@@ -96,6 +94,8 @@
 
     private String _hashmac = GetHashMac.GetHashMac();
 
+    private Boolean firstInstallationByVersion;
```

Comment:
Unboxed `boolean` should be enough.

Reply:
You need a lazy var? Looks like calling `isFirstInstallationByVersion()` is enough.

Reply:
In previous code, `isFirstInstallationByVersion()` always tries to read "data.xml". But "data.xml" will be generated/updated by `initializeTelemetry()`. After this statement, `isFirstInstallationByVersion()` always returns `false`. That's why I add a field member here, to calculate and preserve the real value at the beginning of `initComponent()`.

Reply:
As for the boolean vs Boolean, I need 3 states (null/true/false) to distinguish ( not yet compare data.xml / first-time installation / not first-time ), that's my intention to use the `Boolean`.
I think it's also ok to use the primitive type `boolean`, if we can make sure to assign the correct value to it at the very beginning.

Reply:
Cool, I image it might be better to add keyword `synchronized` to method `isFirstInstallationByVersion()` to avoid double-entry.

Reply:
added